### PR TITLE
fix: snapshots failing because of wrong mysql colum quoting

### PR DESCRIPTION
**Pull request by outinim**
### Description

When I do **snapshots** on a table on which i add columns: it use the function (if I read correctly the file [dbt/include/mariadb/macros/materializations/snapshot/snapshot.sql](https://github.com/dbeatty10/dbt-mysql/blob/cfd275c55adf229d890cd71460bcf520cda5166a/dbt/include/mariadb/macros/materializations/snapshot/snapshot.sql#L62)) `{% do create_columns(target_relation, missing_columns) %}`
which compiles into (for my source table `users` with a new column `team`) into
`alter table `sandbox.users_snapshot` add column "team" varchar(10);`
But it needs to be add column ` team ` to work in mysql 

### Checklist
 - [ ] I have run this code in development and it appears to resolve the stated issue
 - [ ] This PR includes tests, or tests are not required/relevant for this PR
 - [ ] I have updated the `CHANGELOG.md` with information about my change

### CODE REVIEW

File: dbt/include/mariadb/macros/materializations/snapshot/create_columns.sql [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{#
+    Add new columns to the table if applicable
+#}
+{% macro create_columns(relation, columns) %}
```
Comment (mwallace582) [inline dbt/include/mariadb/macros/materializations/snapshot/create_columns.sql:4]: This macro is not necessary, as [it already exists](https://github.com/dbt-labs/dbt-core/blob/1.7.latest/core/dbt/include/global_project/macros/materializations/snapshots/helpers.sql#L5) within the base `dbt-core` SQL templates.

File: dbt/include/mariadb/macros/materializations/snapshot/create_columns.sql
Comment (mwallace582) [inline dbt/include/mariadb/macros/materializations/snapshot/create_columns.sql:1]: While I don't think it's wrong to put this in a new file, I personally would just add the new create_columns macro to the existing `dbt/include/mariadb/macros/materializations/snapshot/snapshot.sql` file.

File: dbt/include/mariadb/macros/materializations/snapshot/create_columns.sql [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  {{ adapter.dispatch('create_columns', 'dbt')(relation, columns) }}
+{% endmacro %}
+
+{% macro default__create_columns(relation, columns) %}
```
Comment (mwallace582) [inline dbt/include/mariadb/macros/materializations/snapshot/create_columns.sql:8]: The `default_` prefix is used in `dbt-core` as a prefix for the default implementation of a macro. Any adapter specific macros should be prefixed with their adapter name (`mariadb_` in this case). 
```suggestion
{% macro mariadb__create_columns(relation, columns) %}
```

File: dbt/include/mariadb/macros/materializations/snapshot/create_columns.sql
Comment (mwallace582) [inline dbt/include/mariadb/macros/materializations/snapshot/create_columns.sql:1]: Due to the way that this repo is structured, when you add something for one adapter flavor (`mariadb`), you should add the same thing for `mysql5` and `mysql` (if applicable).

**Review by mwallace582 — CHANGES_REQUESTED** [review]
Made some comments, hopefully they're helpful. It'd would also be great if you'd be able to add a unit test for this new functionality.

**Comment by outinim** [comment]
Something like this is better ? Thanks for the reviews guys

File: dbt/include/mysql5/macros/materializations/snapshot/snapshot.sql [hunk 3]
```diff
@@ -59,7 +70,7 @@
                                    | rejectattr('name', 'equalto', 'DBT_UNIQUE_KEY')
                                    | list %}
 
-      {% do create_columns(target_relation, missing_columns) %}
+      {% do mysql5__create_columns(target_relation, missing_columns) %}
```
Comment (mwallace582) [inline dbt/include/mysql5/macros/materializations/snapshot/snapshot.sql:73]: This change (and the others like it) is not necessary. The `create_columns()` macro will automatically call the adapter's specific macro if one exists (which it now does).

**Comment by mwallace582** [comment]
Sorry for the delay in taking a look at your changes. This is looking a lot better, and I've just got the one new comment.

It'd also be great if you'd be able to add some tests around this, although I'd be willing to merge this without them given the small scope of the changes.

File: dbt/include/mysql5/macros/materializations/snapshot/snapshot.sql (hunk 3 above)
Comment (mwallace582) [reply dbt/include/mysql5/macros/materializations/snapshot/snapshot.sql]: I've made this change myself so that we can get this merged.

**Pull request merged**